presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -291,6 +291,13 @@ public void testCacheRead(ReadMode readMode)
 
         assertEquals(readFile(cachingFileSystem.open(file)), randomData);
 
+        if (readMode == ASYNC) {
+            // wait for async Rubix requests to complete
+            assertEventually(
+                    new Duration(10, SECONDS),
+                    () -> assertEquals(getAsyncDownloadedMb(readMode), beforeAsyncDownloadedMb + 1));
```

Comment:
why is `getAsyncDownloadedMb` be guaranteed to go up from `beforeAsyncDownloadedMb` to `beforeAsyncDownloadedMb + 1`?

Reply:
> why is getAsyncDownloadedMb be guaranteed to go up from beforeAsyncDownloadedMb to beforeAsyncDownloadedMb + 1

test file has exactly 1MB, so the counter should increase by this much

Reply:
Isn't it possible that counter would flip before whole file is downloaded? (not sure if that matters very much)

Reply:
> Isn't it possible that counter would flip before whole file is downloaded?

It shouldn't happen. In each test we assert that whole file gets downloaded asynchronously before continuing, so there should be no pending counter updates.

Reply:
The observed counter increase can still be from the time when file was still being downloaded. Right

Reply:
> The observed counter increase can still be from the time when file was still being downloaded. Right?

Since the file is 1MB, then when the counter bumps, then file should already be downloaded.